docs/documentation/upgrading/topics/keycloak/changes-24_0_0.adoc

```diff
@@ -174,6 +174,25 @@ The `--proxy` option has been deprecated and will be removed in a future release
 NOTE: For hardened security, the `--proxy-headers` option does not allow selecting both `forwarded` and `xforwarded` values at the same time (as it was
 the case before for `--proxy edge` and `--proxy reencrypt`).
 
+WARNING: When using the proxy headers option, make sure your reverse proxy properly sets and overwrites the `Forwarded` or `X-Forwarded-*` headers respectively.
+To set these headers, consult the documentation for your reverse proxy. Misconfiguration will leave {project_name} exposed to security vulnerabilities.
+
+You can also set the proxy headers when using the Operator:
+[source,yaml]
+----
+apiVersion: k8s.keycloak.org/v2alpha1
+kind: Keycloak
+metadata:
+  name: example-kc
+spec:
+  ...
+  proxy:
+    headers: forwarded|xforwarded
+----
+NOTE: If the `proxy.headers` field is not specified, the Operator falls back to the previous behaviour by implicitly setting
```

Comment:
We would also set it to edge if tls was not configured. Since there isn't a way with the new option to directly specify don't accept anything - should this mention that you may want to specify KC_PROXY=none if tls is not configured? Or do we only expected unencrypted access from behind an edge proxy?

Reply:
> `none` would break the v1 hostname provider

That's also why we can't remove the fallback to `proxy=passthrough` until we've switched to hostname v2.

Reply:
> No, we were not setting proxy=edge from the reasons stated in the PR desc. edge enables parsing the (X-)Forwarded headers which we can't possibly know if they are truly set by the proxy.

We may be miscommunicating. I'm referring to this code: https://github.com/keycloak/keycloak/blob/b1a6a5a1b0e228e59d85a000240e39f13cd5446d/operator/src/main/java/org/keycloak/operator/controllers/KeycloakIngressDependentResource.java#L67 - we are currently setting edge if tls is not configured.

Reply:
> Does that make sense?

Yes